File: backend/QuestionService/src/models/questionModel.ts
import mongoose, { InferSchemaType } from "mongoose";

const Schema = mongoose.Schema;

const questionSchema = new Schema({
    qid: {
        type: Number,
        required: true,
        unique: true
    },
    title: {
        type: String,
        required: true,
        unique: true
    },
    description: {
        type: String,
        required: true
    },
    categories: {
        type: [String],
        required: true
    },
    complexity: {
        type: String,
        enum: ["Easy", "Medium", "Hard"],
        default: "Easy",
        required: true
    }
})
type Question = InferSchemaType<typeof questionSchema>;

export default mongoose.model<Question>("Question", questionSchema);

File: backend/QuestionService/models/questionModel.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Question struct {
	ID          primitive.ObjectID `bson:"_id"`
	Title       string   `bson:"title"`
	Description string   `bson:"description"`
	Categories  []string   `bson:"categories"`
	Complexity  string   `bson:"complexity"`
	Link        string   `bson:"link"`
}

type LeetCodeAPIRequest struct {
	Title string `json:"title"`
}